specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/managedHsm.json

```diff
@@ -348,6 +348,251 @@
           "nextLinkName": "nextLink"
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.KeyVault/managedHSMs/{name}/privateEndpointConnections/{privateEndpointConnectionName}": {
+      "get": {
+        "tags": [
+          "MHSMPrivateEndpointConnections"
+        ],
+        "operationId": "MHSMPrivateEndpointConnections_Get",
+        "description": "Gets the specified private endpoint connection associated with the managed HSM Pool.",
+        "x-ms-examples": {
+          "ManagedHsmGetPrivateEndpointConnection": {
+            "$ref": "./examples/ManagedHsm_getPrivateEndpointConnection.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ManagedHsmResourceGroupName"
+          },
+          {
+            "name": "name",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "Name of the managed HSM Pool"
+          },
+          {
+            "$ref": "#/parameters/MHSMPrivateEndpointConnectionName"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "responses": {
+          "200": {
+            "description": "Private endpoint connection successfully returned.",
+            "schema": {
+              "$ref": "#/definitions/MHSMPrivateEndpointConnection"
+            }
+          },
+          "204": {
```

Comment:
Why not a 404? At least in the data-plane, this is highly unusual.

Reply:
This is as per ARM guidelines for resource-manager. 204 is returned in case Delete has been successful.
https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#delete-resource-asynchronously

Reply:
GET responses following a successful DELETE should be returning 404. Only the DELETE response is to return 204 if the resource didn't exist (it allows idempotent deletes). Please don't implement the current spec'd out behavior.

---
In reply to: [574878958](https://github.com/Azure/azure-rest-api-specs/pull/12938#discussion_r574878958) [](ancestors = 574878958)

Reply:
This is fixed as suggested. Thanks!